#15 improve host quick start (#17)
shinzonetwork/docs: docs/hosts/quickstart.md

description: xxx
---

Hosts transform raw blockchain data into structured **Views** and contribute to network security by producing **Attestation Records**. This quickstart guides you through running the Host Client.

---

## 1. Install DefraDB (Required for Hosting)

Hosts must run a local **DefraDB** instance to sync and store data.

Below are the minimal installation and startup steps extracted from the official documentation.

### Install DefraDB

```bash
git clone https://github.com/sourcenetwork/defradb.git
cd defradb
make install
export PATH=$PATH:$(go env GOPATH)/bin
```

> 

Reviewer: I don't believe this is required since defra is being run as an embedded instance. Did you encounter an error trying to run the host without having defra installed?

Author: ah I must have hit a different error and confused it as a defradb requirement. I'm removing defraDB setup from the docs

### Key Management (Initial Setup)

Create a key for signing operations:

```bash
export DEFRA_KEYRING_SECRET=<make_a_password>
defradb keyring generate

> ```

Reviewer: Definitely not needed

### Start DefraDB

Start a local node:

```bash
defradb start
```

This will expose the DefraDB API at the default local port.

---

> 

Reviewer: Not needed

Reviewer: Hosts should not be running standalone instances of defra

## 2. Install the Shinzo Host Client

Clone the repository and enter the directory:

```bash
git clone https://github.com/shinzonetwork/shinzo-host-client.git
cd shinzo-host-client
```

---

## 3. Configuration

The Host Client reads from `config.yaml`, which comes with sensible defaults.
Only modify values when needed.

### Key Fields

* **defradb.url** – API endpoint of your local DefraDB node. Defaults work for most setups.

> * **defradb.keyring_secret** – Optional. Usually provided via the `DEFRA_KEYRING_SECRET` environment variable.

Reviewer: Not optional. You must provide a keyring secret in either an env var or in config

* **p2p.bootstrap_peers** – Seed peers for joining the Shinzo network. Defaults include a reliable bootstrap peer.
* **p2p.listen_addr** – Default is suitable for local runs. Override when containerizing.
* **store.path** – Directory where local DefraDB data is stored.
* **shinzo.web_socket_url** – Defaults to a hosted ShinzoHub node. Only change if connecting to a different node.
* **logger.development** – Set to `false` for production.
* **host.lens_registry_path** – Where received WASM lens files are stored.

### Default Behavior

The included `config.yaml` is ready for most local development workflows. You should only need to modify peer settings or storage paths for advanced setups.

---

## 4. Running the Host

To start the Host Client:

```bash
go run cmd/main.go
```

### Or Run with Playground Enabled

```bash
make build-with-playground
make start-playground
```

This runs the Host and also exposes a **Playground GUI** at:

```
http://192.168.50.33:9182

> ```

Reviewer: I don't believe this is correct

Author: ah yeah that's my LAN IP. Have you been able to access the playground from localhost or 127.0.0.1? I haven't been able to. I'm editing the docs to instruct users to look for `🧪 GraphQL Playground available at http://192.168.50.33:9182`

> description: xxx

Reviewer 2: let's add description here "Run a Shinzo Host to transform blockchain data into verifiable Views"

---

> Hosts transform raw blockchain data into structured **Views** and contribute to network security by producing **Attestation Records**. This quickstart guides you through running the Host Client.

Reviewer 2: ```suggestion
Hosts transform raw blockchain data into structured **Views** and contribute to network security by producing **Attestation Records**. This quick start guide walks you through installing, configuring, and running the Shinzo Host Client.
```

---

## 1. Install the Shinzo Host Client

Clone the repository and enter the directory:

```bash
git clone https://github.com/shinzonetwork/shinzo-host-client.git
cd shinzo-host-client
```

---

## 2. Configuration

> The Host Client reads from `config.yaml`, which comes with sensible defaults.

Reviewer 2: ```suggestion
The Host Client reads from [config.yaml](https://github.com/shinzonetwork/shinzo-host-client/blob/main/config.yaml) which comes with sensible defaults.
```

The only field you need to set is **defradb.keyring_secret** which can alternatively be set with the following command in the terminal window.

```bash
export DEFRA_KEYRING_SECRET=<make_a_password>
```

### Key Fields

* **defradb.url** – API endpoint of your local DefraDB node. Defaults work for most setups.
* **defradb.keyring_secret** – Requires a secret to generate your private keys. 
* **p2p.bootstrap_peers** – Seed peers for joining the Shinzo network. Defaults include a reliable bootstrap peer.
* **p2p.listen_addr** – Default is suitable for local runs. Override when containerizing.
* **store.path** – Directory where local DefraDB data is stored.
* **shinzo.web_socket_url** – Defaults to a hosted ShinzoHub node. Only change if connecting to a different node.
* **logger.development** – Set to `false` for production.

> * **host.lens_registry_path** – Where received WASM lens files are stored.

Reviewer 3: suggestion: explain how to connect to a local indexer. Right now, if I follow the indexer getting started guide and then host quickstart, I get all sorts of errors. For example, both Indexer and Host want to start a server on 9181 port and it prevents host from starting it. Setting `defradn.url` to a different port doesn't mean it won't try to set up 9181 port. 

Also, following these guides, ShinzoHub node is probably not set up yet, so `shinzo.web_socket_url` is not yet ready. It would be nice to link the field to a guide on setting up ShinzoHub

Author: Good point about the local indexer. Let me go through those docs and see how to make the hosts docs compatible.

The websocket url will be updated when this PR is merged: https://github.com/shinzonetwork/shinzo-host-client/pull/57

### Default Behavior

The included `config.yaml` is ready for most local development workflows. You should only need to modify peer settings or storage paths for advanced setups.

> ---

Reviewer 2: ```suggestion
```

can we remove these lines from the guide to be consistent across other files.

## 4. Running the Host

> To start the Host Client:

Reviewer 2: @akc2267 Can we add a note here clarifying that the localhost WebSocket URL is only for users running a local ShinzoHub? We should also mention that this value can be replaced with the deployed ShinzoHub endpoint when not running locally.

Author: Duncan said we don't want users to run a local ShinzoHub node. the localhost websocket url will be replaced in this PR https://github.com/shinzonetwork/shinzo-host-client/pull/57
